sig/src/Decon2DFilter.cxx

```diff
@@ -0,0 +1,165 @@
+#include "WireCellSig/Decon2DFilter.h"
+#include "WireCellSig/Util.h"
+
+#include "WireCellUtil/NamedFactory.h"
+#include "WireCellUtil/String.h"
+#include "WireCellUtil/Array.h"
+#include "WireCellUtil/Response.h"
+#include "WireCellUtil/FFTBestLength.h"
+#include "WireCellUtil/Exceptions.h"
+
+#include "WireCellIface/ITensorSet.h"
+#include "WireCellIface/IFilterWaveform.h"
+
+#include "WireCellAux/SimpleTensorSet.h"
+#include "WireCellAux/SimpleTensor.h"
+#include "WireCellAux/Util.h"
+#include "WireCellAux/TensUtil.h"
+
+WIRECELL_FACTORY(Decon2DFilter, WireCell::Sig::Decon2DFilter, WireCell::ITensorSetFilter, WireCell::IConfigurable)
+
+using namespace WireCell;
+
+Sig::Decon2DFilter::Decon2DFilter()
+  : log(Log::logger("sig"))
+{
+}
+
+Configuration Sig::Decon2DFilter::default_configuration() const
+{
+    Configuration cfg;
+
+    return cfg;
+}
+
+void Sig::Decon2DFilter::configure(const WireCell::Configuration &cfg) { m_cfg = cfg; }
+
+bool Sig::Decon2DFilter::operator()(const ITensorSet::pointer &in, ITensorSet::pointer &out)
+{
+    log->debug("Decon2DFilter: start");
+
+    if (!in) {
+        out = nullptr;
+        return true;
+    }
+    auto set_md = in->metadata();
+
+    auto tag = get<std::string>(m_cfg, "tag", "trace_tag");
+
+    auto iwf_ten = Aux::get_tens(in, tag, "waveform");
+    if (!iwf_ten) {
+        log->error("Tensor->Frame: failed to get waveform tensor for tag {}", tag);
+        THROW(ValueError() << errmsg{"Sig::Decon2DFilter::operator() !iwf_ten"});
+    }
+    auto wf_shape = iwf_ten->shape();
+    size_t nchans = wf_shape[0];
+    size_t nticks = wf_shape[1];
+    log->debug("iwf_ten->shape: {} {}", nchans, nticks);
+
+    auto ch_ten = Aux::get_tens(in, tag, "channels");
+    if (!ch_ten) {
+        log->error("Tensor->Frame: failed to get channels tensor for tag {}", tag);
+        THROW(ValueError() << errmsg{"Sig::Decon2DFilter::operator() !ch_ten"});
+    }
+    Eigen::Map<const Eigen::ArrayXi> ch_arr((const int *) ch_ten->data(), nchans);
+
+    bool have_cmm = true;
+    auto cmm_range = Aux::get_tens(in, tag, "bad:cmm_range");
```

Comment:
The "bad" part of this string should be configurable with a default value of "bad".  This need not be done immediately.